cypress-io/cypress
# chore: adding beforeTest and AfterTest to protocol

### Additional details


This PR adds the `test:lifecycle` `start` and `end` events to the protocol.

### Steps to test


- Clone the `cypress-services` repo and checkout `mschile/protocol/add_before_after_test`
  - Run `yarn`
  - Run `yarn watch` in `packages/app-capture-protocol`
- Set `CYPRESS_LOCAL_PROTOCOL_PATH` to the path to `cypress-services/packages/app-capture-protocol/dist/index.js`
- Clone the `cypress` repo  and checkout `mschile/protocol/before_after_test`
  - Run `yarn`
  - Execute `cypress run` on a project in record mode with `DEBUG` set to `cypress:*protocol*` and ensure the `test:lifecycle` events are added to the database.

### How has the user experience changed?

<img width="1063" alt="Screenshot 2023-03-24 at 12 19 24 PM" src="https://user-images.githubusercontent.com/2002044/227608024-127dc2f3-392c-4a54-b7e7-684cbc79ee78.png">

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
